src/iris/api.py

```diff
@@ -1999,13 +2016,54 @@ def on_post(self, req, resp):
             ]
 
         This will map target 0 to the user "jdoe", and target 1 to the team "team-foo".
+
+        To use dynamic tracking you can pass a dynamic_tracking_notifications list in the incident_params. If the plan has dynamic tracking enabled, the incident will be created with the dynamic tracking notifications specified in the incident_params. The tracking message will use the tracking template specified in the plan.
+
+        .. sourcecode:: json
+
+        "dynamic_tracking_notifications": [{"mode": "slack", "destination": "#iris-slack-testing"}, {"mode": "slack", "destination": "#iris-slack-testing2"}]
         '''
-        incident_params = ujson.loads(req.context['body'])
+        try:
+            incident_params = ujson.loads(req.context["body"])
+        except ValueError:
+            raise falcon.HTTPBadRequest(
+                "Invalid JSON", "Could not parse the request body as JSON."
+            )
         dynamic_targets = []
-        if 'plan' not in incident_params:
-            raise HTTPBadRequest('missing plan name attribute')
+        if "plan" not in incident_params:
+            raise HTTPBadRequest("missing plan name attribute")
 
-        app = req.context['app']
+        app = req.context["app"]
+
+        dynamic_tracking_notifications = incident_params.get(
+            "dynamic_tracking_notifications", []
+        )
+        if not isinstance(dynamic_tracking_notifications, list):
+            raise falcon.HTTPBadRequest(
+                "Invalid Format", "dynamic_tracking_notifications should be a list."
+            )
+        for notification in dynamic_tracking_notifications:
+            if not isinstance(notification, dict):
```

Comment:
At some point, using a library like marshmallow would help simplify this greatly by avoiding all of these manual checks

Reply:
yeah, we could definitely benefit massively from adding that style of validation to all endpoints